[2017.7] Merge forward from 2016.11 to 2017.7

rallytime opened the pull request:
Conflicts:
  - salt/modules/win_pkg.py
  - salt/states/file.py
  - tests/integration/states/test_file.py
  - tests/unit/states/test_file.py

Commit:
Bump deprecation warnings from Oxygen to Fluorine

The following files had deprecation warnings for "Oxygen" that should
be handled in the next feature release (Fluorine):

- salt/client/mixins.py
- salt/states/lxc.py
- salt/utils/__init__.py

gtmanfred committed:
if gateway is not specified use iface

This will allow for the route to be applied to drop the packets on that
interface.

thatch45 committed:
Merge pull request #44741 from gtmanfred/rhip

if gateway is not specified use iface

thatch45 committed:
Merge pull request #44738 from rallytime/bump-oxygen-warnings

Bump some deprecation warnings from Oxygen to Fluorine

terminalmage committed:
Fix regression in file.managed when source_hash used with local file

This ensures that we enforce the passed source_hash when the source is a
file local to the minion.

Commit:
Remove debugging line

Commit:
Merge pull request #44794 from terminalmage/issue44365

Fix regression in file.managed when source_hash used with local file

Commit:
Fix spelling typo, and fix backwards campatible minion option for repo location

Commit:
Lint: Remove extra whitespace

Commit:
Merge pull request #44852 from damon-atkins/2016.11_win_pkg_typo_n_fix

win_pkg fix spelling typos and minion option 2016.11

Commit:
Merge branch '2016.11' into '2017.7'

Conflicts:
  - salt/modules/win_pkg.py
  - salt/states/file.py
  - tests/integration/states/test_file.py
  - tests/unit/states/test_file.py

damon-atkins commented:
If you make the following updates, my 2017.11 merge should not be required.

in def list_available(*names, **kwargs) the following matches 2016.11

```
    saltenv = kwargs.get('saltenv', 'base')
    refresh = salt.utils.is_true(kwargs.get('refresh', False))
    _refresh_db_conditional(saltenv, force=refresh)
    return_dict_always = \
        salt.utils.is_true(kwargs.get('return_dict_always', False))
    if len(names) == 1 and not return_dict_always:
```


The following matches 2016.11, not show why this did not come from 2016.11
```
def list_pkgs(versions_as_list=False, **kwargs):
    '''
    List the packages currently installed
    Args:
        version_as_list (bool): Returns the versions as a list

    Kwargs:
```
The following should be the only changes in the code between 2016 and 2017

- 2017 introduce `salt.utils.path.safe_path()` which replaced some code.
- 2017 moved some functions from `utils/__init__.py` to `utils/something.py`
- 2017 removes the warning `if 'win_repo_source_dir' in __opts__: salt.utils.warn_until('Nitrogen',` in `_get_repo_details()`

Commit:
Lint: Remove uses of integration.TMP and integration.FILE

This path is no longer used in 2017.7.

rallytime commented:
@damon-atkins Thanks for your feedback, but I won't be making those changes in this pull request. We like to try and keep them as simple as possible. Changes like that can be made in a separate PR.

Commit:
Fix test failure: the new comment to test against moved to correct place

The original conflict resolution didn't place the updated comment in
the right spot.

rallytime commented:
This has conflicts, so I will redo it.
